### PR TITLE
Refactor binary release

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/app/models/binary_release.rb [hunk 1]
```diff
@@ -47,14 +47,15 @@ def self.update_binary_releases_via_json(repository, json, time = Time.now)
...
         # double definition means broken DB entries
-        existing.offset(1).destroy_all
+        matching_binaries.offset(1).destroy_all
 
         # compare with existing entry
-        if existing.count == 1
-          entry = existing.first
+        entry = matching_binaries.first
+
+        if entry
```
Comment (ChrisBr) [inline src/api/app/models/binary_release.rb]: [nitpick] entry.present?

File: src/api/app/models/binary_release.rb [hunk 2]
```diff
@@ -187,7 +187,7 @@ def indentical_to?(binary_hash)
     binary_disturl == binary_hash['disturl'] &&
       binary_supportstatus == binary_hash['supportstatus'] &&
       binary_buildtime &&
-      binary_buildtimeto.datetime.utc == ::Time.at(binary['buildtime'].to_i).to_datetime.utc
+      binary_buildtime.datetime.utc == ::Time.at(binary['buildtime'].to_i).to_datetime.utc
```
Comment (ChrisBr) [inline src/api/app/models/binary_release.rb]: I think it is not a type it was meant to be ``binary_buildtime_TO.datetime.utc``
But anyway, I'm fine with the change!

File: src/api/app/models/binary_release.rb (hunk 1 above)
Comment (bgeuken) [reply src/api/app/models/binary_release.rb]: `matching_binaries` is an ActiveRecord query result. So `matching_binaries.first` is either nil or a binary release object

File: src/api/app/models/binary_release.rb (hunk 2 above)
Comment (bgeuken) [reply src/api/app/models/binary_release.rb]: Nah, it should be to_datetime :/

**Pull request merged**